src/NotepadNext/ScintillaNext.cpp

```diff
@@ -597,3 +599,260 @@ void ScintillaNext::setTemporary(bool temp)
     // Fake this signal
     emit savePointChanged(temporary);
 }
+
+void ScintillaNext::keyPressEvent(QKeyEvent *event)
```

Comment:
Another issue is this is enabled for all files so could inadvertently get triggered. It should only be enabled for plain text files.

Reply:
This idea is too conservative. Let's first look at the possible problems of global activation, and then consider how to avoid them.
First of all, real-time evaluation is non-invasive. Because the text is not modified, there is no need to narrow the scope of use. If it is really not needed, you can turn it off in the settings.
And `=ENTER` and `=?` may insert a number up to 15 characters in length in the text. This can be harmful in some situations, however it can also be useful and can be limited to avoid accidental presses.
Just give an example of an application scenario. An embedded developer can write it in a comment to calculate the address offset and save it with the source code. Since the calculation formula in the comment is still there, the next time it needs to be recalculated, it can be directly added in the comment. Evaluate after modifying the value.
So I think we can set up a whitelist according to document types. The file types in the whitelist can be used without restrictions. For those outside the whitelist, the operation method of intrusive `=` sign evaluation will be changed. I think using `=Shift+ ENTER` replaces `=ENTER`, and `=Ctrl+?` replaces `=?`.
What do you think of it?